skops/io/tests/test_persist.py

```diff
@@ -143,13 +174,9 @@ def assert_params_equal(params1, params2):
         val1, val2 = params1[key], params2[key]
         assert type(val1) == type(val2)
 
-        if (
-            (key == "steps")
-            or (key == "transformer_list")
-            or key.endswith("__steps")
-            or key.endswith("__transformer_list")
-        ):
-            # we deal with a pipeline or feature union
+        if _is_steps_like(val1):
+            # Deal with Pipeline.steps, FeatureUnion.transformer_list, etc.
+            assert _is_steps_like(val2)
```

Comment:
are we also checking their contents?

Reply:
Yes, we transform them to dicts and below we compare the dicts.